src/spec-configuration/containerTemplatesOCI.ts

```diff
@@ -2,31 +2,52 @@ import { Log, LogLevel } from '../spec-utils/log';
 import * as os from 'os';
 import * as path from 'path';
 import { fetchOCIManifestIfExists, getBlob, getRef, OCIManifest } from './containerCollectionsOCI';
+import { isLocalFile, readLocalFile, writeLocalFile } from '../spec-utils/pfs';
 
+export interface TemplateOptions {
+	[name: string]: string;
+}
 
-export async function fetchTemplate(output: Log, identifier: string, templateCachePath: string): Promise<string[] | undefined> {
-	const templateRef = getRef(output, identifier);
+export interface SelectedTemplate {
+	id: string;
+	options: TemplateOptions;
+}
+
+export async function fetchTemplate(output: Log, selectedTemplate: SelectedTemplate, templateDestPath: string): Promise<string[] | undefined> {
+	const { id, options } = selectedTemplate;
+	const templateRef = getRef(output, id);
 	if (!templateRef) {
-		output.write(`Failed to parse template ref for ${identifier}`, LogLevel.Error);
+		output.write(`Failed to parse template ref for ${id}`, LogLevel.Error);
 		return;
 	}
 
-	const ociManifest = await fetchOCITemplateManifestIfExistsFromUserIdentifier(output, process.env, identifier);
+	const ociManifest = await fetchOCITemplateManifestIfExistsFromUserIdentifier(output, process.env, id);
 	if (!ociManifest) {
-		output.write(`Failed to fetch template manifest for ${identifier}`, LogLevel.Error);
+		output.write(`Failed to fetch template manifest for ${id}`, LogLevel.Error);
 		return;
 	}
 
 	const blobUrl = `https://${templateRef.registry}/v2/${templateRef.path}/blobs/${ociManifest?.layers[0].digest}`;
 	output.write(`blob url: ${blobUrl}`, LogLevel.Trace);
 
 	const tmpDir = path.join(os.tmpdir(), 'vsch-template-temp', `${Date.now()}`);
-	const files = await getBlob(output, process.env, blobUrl, tmpDir, templateCachePath, templateRef, undefined, ['devcontainer-template.json', 'README.md', 'NOTES.md']);
+	const files = await getBlob(output, process.env, blobUrl, tmpDir, templateDestPath, templateRef, undefined, ['devcontainer-template.json', 'README.md', 'NOTES.md']);
 
 	if (!files) {
 		throw new Error(`Failed to download package for ${templateRef.resource}`);
 	}
 
+	// Scan all template files and replace any templated values.
+	for (const f of files) {
```

Comment:
Would this scan and replace recursively? (ie through the sub-folders)

Reply:
Yes it does (anaconda test proves this)